src/features/machines/swapIntentMachine.ts

```diff
@@ -501,14 +502,37 @@ export const swapIntentMachine = setup({
           ],
         },
 
-        onDone: {
-          target: "Completed",
-
-          actions: {
-            type: "setIntentHash",
-            params: ({ event }) => event.output,
+        onDone: [
+          {
+            target: "Completed",
+            guard: {
+              type: "isOk",
+              params: ({ event }) => event.output,
+            },
+            actions: {
+              type: "setIntentHash",
+              params: ({ event }) => event.output.value as string,
+            },
           },
-        },
+          {
+            target: "Generic Error",
+            actions: [
+              {
+                type: "logError",
```

Comment:
I doubt we need to log this, as it's not exception, it's something that could happen and we expect it.